[Chen Xiangyun] iP

chenxy12345 opened the pull request.

Commit:
Add Gradle support

chenxy12345 committed:
Rename, Greet, Exit

chenxy12345 committed:
Echo

chenxy12345 committed:
Add, List

chenxy12345 committed:
Revert "Add, List"

This reverts commit 137005fbf90d410ebc3e76dacac3cee2f4134ad7.

chenxy12345 committed:
Add, List

chenxy12345 committed:
Mark as Done, added Task.java

chenxy12345 committed:
Todo, Event, Deadline

chenxy12345 committed:
Automated Text UI Testing

chenxy12345 committed:
Handle Errors

chenxy12345 committed:
Minor Changes

chenxy12345 committed:
Add Delete

chenxy12345 committed:
Add save

chenxy12345 committed:
Add Dates and Times

chenxy12345 committed:
Add OOP

chenxy12345 committed:
Add Packages

chenxy12345 committed:
Merge commit '68c58c12c1d631736a4b25500248a2956171496f'

* commit '68c58c12c1d631736a4b25500248a2956171496f':
  Add Gradle support

chenxy12345 committed:
Add gradle

chenxy12345 committed:
Add JUnit Tests

chenxy12345 committed:
Add JavaDoc Comments

chenxy12345 committed:
Add CodingStandard

chenxy12345 committed:
Add Find

chenxy12345 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Add CodingStandard

chenxy12345 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add Find

mogmyij commented on src/main/java/Elmacho.java:
```diff
@@ -0,0 +1,60 @@
+import task.Tasklist;
```
I think that this should be part of a package based on the coding conventions all Classes should belong to a package

boinkkitty replied:
May consider creating a package for this to follow the coding standards

mogmyij commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,76 @@
+
+    public void start() {
+        String name = "Elmacho";
+        System.out.println("____________________________________________________________");
```
Maybe this could be stored as a constant variable instead which would allow for changes to be made easier

boinkkitty replied:
Could declare it as a start as a constant to make it more convenient and prevent human error

mogmyij commented on src/main/java/exceptions/ElmachoExceptions.java:
```diff
@@ -0,0 +1,9 @@
+package exceptions;
+
+public class ElmachoExceptions extends Exception{
```
I think by the coding quality conventions plural forms of words should be used for multi-valued variables like ArrayLists

mogmyij commented on src/main/java/task/Tasklist.java:
```diff
@@ -0,0 +1,86 @@
+    private int numberOfTasks;
+
+    public Tasklist() {
+        this.arrayA = new Task[100];
```
maybe use ArrayList to avoid having to resize the array

mogmyij reviewed (commented):
Looks good to me 

just some small comments

boinkkitty commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,15 @@
+import ui.Ui;
+
+
+public class Command {
```
Could consider making this class and execute method abstract if appropriate

boinkkitty commented on src/main/java/exceptions/ElmachoExceptions.java:
```diff
@@ -0,0 +1,9 @@
+package exceptions;
+
```
May add brief description of class

boinkkitty commented on src/main/java/task/Tasklist.java:
```diff
@@ -0,0 +1,86 @@
+     * @param n The index of the task to be deleted.
+     * @throws ElmachoExceptions if the index is not on the tasklist.
+     */
+    public void delete(int n) throws ElmachoExceptions {
```
Could consider using ArrayList to omit shifting of elements

boinkkitty commented on src/main/java/task/Tasklist.java:
```diff
@@ -0,0 +1,86 @@
+            this.arrayA[n].unmark();
+        }
+    }
+
```
Perhaps could add some documentation

boinkkitty commented on src/main/java/task/ToDo.java:
```diff
@@ -0,0 +1,21 @@
+        if (super.isDone) {
+            return "T " + 1 + "/" + description;
+        }
+        return "T " + 0 + "/" + description;
```
I think adding an 'else' block would be closer to coding conventions

boinkkitty commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,37 @@
+package task;
+
```
Could consider making it abstract

boinkkitty commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,37 @@
+        if (isDone) {
+            return "E " + 1 + "/" + description + "/from" + from.format(formatter)
+                    + " to" + to.format(formatter);
+        }
```
'Else' statement might be needed to follow coding conventions

boinkkitty commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,139 @@
+        String[] parts = instruction.split(" ", 2);
+        String command = parts[0];
+
+        if (command.equals("delete")) {
```
Could consider using switch-case if it's better for you

boinkkitty reviewed (commented):
Overall well done! Just a bit more documentation and not too sure about whether not using "else if()" follows coding convention

WhyAre commented on src/main/java/Elmacho.java:
```diff
@@ -0,0 +1,60 @@
+        Scanner scanner = new Scanner(System.in);
+        Parser parser = new Parser();
+
+        while (true) {
```
Would it be better to use a `do-while` loop?

WhyAre commented on src/main/java/command/DeleteCommand.java:
```diff
@@ -0,0 +1,44 @@
+ */
+public class DeleteCommand extends Command {
+
+    private int n;
```
Perhaps use a more descriptive name?

WhyAre commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,139 @@
+        // Loading of tasks
+        if (command.equals("T") || command.equals("D") || command.equals("E")) {
+            String[] details = parts[1].split("/");
+            boolean isDone = details[0].equals("1") ? true : false;
```
```suggestion
            boolean isDone = details[0].equals("1");
```
It is sufficient to write this line as follows

WhyAre commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,139 @@
+                }
+                if (command.equals("deadline")) {
+                    String[] deadlineDetails = parts[1].split("/by ", 2);  // Split on "/by " with a limit of 2
+                    if (deadlineDetails.length < 2) {
```
Is it possible to avoid such deep nesting?

chenxy12345 committed:
Changed to ArrayList

chenxy12345 committed:
Add Checkstyle

chenxy12345 committed:
Add GUI

chenxy12345 committed:
Small edits

chenxy12345 committed:
Assertion: to make debugging easier by verifying
assumptions during development

Many assumptions are made in the code.

Assertions are input in place so that the internal assumptions
and conditions that should always hold true are being checked.

Let's,
*update and edit the important assumptions points that
should hold at various points in the code

chenxy12345 committed:
Changed Parser Class parse() method

chenxy12345 committed:
Making the printing of the list a stream

Iterating through the whole Arraylist manually is possible,
but Streams improve the code readability, making the
code more straightforward.

chenxy12345 committed:
Merge pull request #2 from chenxy12345/branch-A-Assertions

Adding Assertions

chenxy12345 committed:
Resolve merge conflict, deleting find() in Tasklist

chenxy12345 committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Assertion: to make debugging easier by verifying assumptions during development

chenxy12345 committed:
Merge pull request #3 from chenxy12345/branch-A-Streams

Adding Streams

chenxy12345 committed:
Add Archived

chenxy12345 committed:
Update GUI

chenxy12345 committed:
Add Motivation

chenxy12345 committed:
Add more Error Handling

chenxy12345 committed:
Add User Guide

chenxy12345 committed:
Refinements to README

chenxy12345 committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:
  Changed Parser Class parse() method

chenxy12345 committed:
Update README.md

chenxy12345 committed:
Update README.md

chenxy12345 committed:
Update README.md

chenxy12345 committed:
Add Minor Updates

chenxy12345 committed:
Merge branch 'master' of https://github.com/chenxy12345/ip

* 'master' of https://github.com/chenxy12345/ip:
  Update README.md
  Update README.md
  Update README.md

chenxy12345 committed:
Update ReadME

chenxy12345 committed:
Updated Storage

chenxy12345 committed:
Updated READMe

chenxy12345 committed:
Update READMe

chenxy12345 committed:
Add handling of user inputs

chenxy12345 committed:
Update README.md

chenxy12345 committed:
Update README.md

chenxy12345 committed:
Update README.md

chenxy12345 committed:
Update README.md

chenxy12345 committed:
Update checksum

chenxy12345 committed:
Update README.md

chenxy12345 committed:
Update README.md

chenxy12345 committed:
Update input handling

chenxy12345 committed:
Update dependencies
